### PR TITLE
[TOOLS-1961] Add budget information to the project - FE

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestJpaRepository.java [hunk 1]
```diff
@@ -39,4 +41,17 @@ public void remove(ChangeRequest item) {
...
 
+    @Override
+    public Optional<ChangeRequest> getBaseline(ProjectFilterConfiguration project){
+        
+        try{
+            
+            return Optional.of(entityManager.createNamedQuery("ChangeRequest.getBaseline", ChangeRequest.class)
+                .setParameter("project", project)
+                .getSingleResult());
+        } catch (NoResultException e){
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestJpaRepository.java]: MAJOR Either log or rethrow this exception. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/objective/taskboard/project/config/ProjectConfigurationController.java [hunk 2]
```diff
@@ -60,7 +64,10 @@
             return ResponseEntity.notFound().build();
 
         ProjectFilterConfiguration project = optConfiguration.get();
-
+        
+        boolean updateBaseline = project.getStartDate().isPresent() && !project.getStartDate().get().equals(configDto.startDate)
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/project/config/ProjectConfigurationController.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 3]
```diff
@@ -1,31 +1,110 @@
...
+            PROJECT_ADMINISTRATION);
+
+        if (!project.isPresent())
+            throw new AccessDeniedException("The current user doesn't have administrative permissions for this project."); 
+
+        ProjectFilterConfiguration projectObj = project.get();
+        Optional<ChangeRequest> existentBaseline = changeRequestRepository.getBaseline(projectObj);
+
+        if (existentBaseline.isPresent()) {
+            existentBaseline.get().setDate(projectObj.getStartDate().get());
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 4]
```diff
@@ -1,31 +1,110 @@
...
+        } else {
+            ChangeRequest baseline = new ChangeRequest(projectObj, "Baseline", projectObj.getStartDate().get(), 0, true);
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 5]
```diff
@@ -1,31 +1,112 @@
...
-            throw new ChangeRequestBaselineRemovalException();
+
+    @Transactional
+    public void updateItems(String projectKey, List<ChangeRequest> requestCrs) {
+
+        Optional<ProjectFilterConfiguration> project = authorizedProjectsService.getTaskboardProject(projectKey,
+            PROJECT_ADMINISTRATION);
+
+        if (!project.isPresent())
+            throw new AccessDeniedException("The current user doesn't have administrative permissions for this project.");
```
[the remark this answers is not in the document]
Comment (taksan) [reply src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: projectAdministrationPermission.isAuthorizeFor(project.getProjectKey())

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 6]
```diff
@@ -1,31 +1,112 @@
...
+        deleteMissingItems(requestCrs, existingCrs);
+    }
+
+    @Transactional
+    public void updateBaselineChangeRequest(String projectKey, LocalDate startDate){
+        Optional<ProjectFilterConfiguration> project = authorizedProjectsService.getTaskboardProject(projectKey,
+            PROJECT_ADMINISTRATION);
+
+        if (!project.isPresent())
+            throw new AccessDeniedException("The current user doesn't have administrative permissions for this project.");
```
[the remark this answers is not in the document]
Comment (taksan) [reply src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: projectAdministrationPermission.isAuthorizeFor(project.getProjectKey())

**Pull request merged**